app/helpers/application_helper.rb

```diff
@@ -9,13 +9,20 @@ def configured_organization_name
   def parse_application_data(data)
     questions = []
     data.each do |k, v|
-      match_data = k.match(/^(prompt|response|data_type)_(\d+)$/)
+      match_data = k.match(/^(prompt|response|data_type)?_?(\d+)$/)
       next unless match_data.present?
-      input_type, number = match_data.captures
-      questions[number.to_i] ||= []
-      input_types = %w(prompt response data_type)
-      index = input_types.index input_type
-      questions[number.to_i][index] = v
+      captures = match_data.captures.select{|c| !c.nil? }
+      if captures.length == 2
+        input_type, number = match_data.captures
+        questions[number.to_i] ||= []
+        input_types = %w(prompt response data_type)
+        index = input_types.index input_type
+        questions[number.to_i][index] = v
+      elsif captures.length == 1
+        q_id, number = v.split('-')
+        questions[number.to_i] ||= []
+        questions[number.to_i][3] = q_id.to_i + 1
```

Comment:
That's a good point! We're never going to come up with a perfect solution in a data retrieval situation like this anyway. The good thing about your solution is that it has a very good chance of working well if the past application was recent, and a lower chance of working well with less recent applications. You've convinced me.